Merge C4DT's developments into C4DT branch

PascalinDe opened the pull request:
This PR is to merge our developments into our dedicated branch on upstream. This is part of a 3-step process.

1. merge our developments into our dedicated branch `c4dt-main`
2. merge our dedicated branch into upstream's `main` branch
3. create a branch on top of `c4dt-main` with the results from the access control student project

PascalinDe committed:
refactor: use Title object instead of MainTitle, TitleDe and TitleFr

PascalinDe committed:
refactor: use/clean up Hint object

PascalinDe committed:
fix: remove internal ChoicesMap from exported JSON

PascalinDe committed:
fix: lint mock data

PascalinDe committed:
fix: run prettier

PascalinDe committed:
fix: fix Go tests

PascalinDe committed:
refactor: use internationalize function

PascalinDe committed:
refactor: remove debugging

PascalinDe committed:
refactor: remove dead code

dragonleman committed:
Fix French lexicon

PascalinDe committed:
fix: updated local_forms script w/ current data format

PascalinDe committed:
tests: add German translations

ineiti committed:
Merge pull request #61 from c4dt/fix_tests

Using correct URL

ineiti committed:
Correctly create new keys

The new keys couldn't be created when the backend was not running.
Fixed it by calling the backend only when necessary.

dragonleman committed:
Décrypt* -> Déchiffr*

PascalinDe committed:
Merge pull request #63 from c4dt/fix-fr-language-syntaxes

Fix French lexicon

PascalinDe committed:
Merge pull request #51 from c4dt/316

fix: repair JSON export

ineiti committed:
Merge pull request #64 from c4dt/fix_scripts

Correctly create new keys

dragonleman committed:
Homogeinization of acronyms

dragonleman committed:
Merge pull request #65 from c4dt/fix-languages-homogeneization-acronyms

Homogeinization of acronyms

PascalinDe committed:
test: install and initialize Playwright

PascalinDe committed:
test: add initalizing i18next

PascalinDe committed:
test: add login/logout utils

PascalinDe committed:
fix: add missing import

PascalinDe committed:
test: add navigation bar tests for authenticated/admin users

PascalinDe committed:
fix: remove duplicate logo

PascalinDe committed:
test: start each test at 'About' page

PascalinDe committed:
test: added test for link to form list

PascalinDe committed:
docs: added README for automated tests

PascalinDe committed:
fix: do not include SCIPER_ADMIN in filename as it might be different for other devs

PascalinDe committed:
fix: clean up Playwright config

PascalinDe committed:
fix: use fixed admin sciper for mocks

PascalinDe committed:
fix: add missing API mock

PascalinDe committed:
docs: updated README w/ installation instructions and required environment variables

PascalinDe committed:
fix: remove duplicate hidden element

PascalinDe committed:
WIP: fix log in/log out mocks

PascalinDe committed:
fix: fix log in/log out helper functions

PascalinDe committed:
docs: add mandatory FRONT_END_URL value to READMEs

PascalinDe committed:
test: clean up authentication mocks

PascalinDe committed:
test: clean up HAR file management and authentication mocking

PascalinDe committed:
test: add authentication UI tests

PascalinDe committed:
docs: add updating HAR files to README

PascalinDe committed:
fix: apply code review suggestions

PascalinDe committed:
test: do not test WebKit

PascalinDe committed:
test: use hard-coded baseURL instead of local configuration

PascalinDe committed:
test: apply review suggestions

PascalinDe committed:
fix: re-add accidentally removed mockPersonalInfo in setUp function

PascalinDe committed:
fix: use regexp for local configuration

PascalinDe committed:
docs: fix npm command in README

PascalinDe committed:
docs: fix npx command

PascalinDe committed:
Merge pull request #67 from c4dt/playwright

Add automated tests for navigation bar

PascalinDe committed:
test: move updating mocks toggle to setUp

PascalinDe committed:
test: add missing skip instructions

PascalinDe committed:
test: add testid to footer

PascalinDe committed:
feat: render copyright notice in footer internationalisable

PascalinDe committed:
feat: remove unused proxy configuration element

PascalinDe committed:
Merge pull request #74 from c4dt/playwright

test: add missing skip instructions

PascalinDe committed:
feat: internationalise build/software information and add tests

PascalinDe committed:
refactor: rename file

PascalinDe committed:
Merge pull request #75 from c4dt/69

Footer UI tests

PascalinDe committed:
Revert "test: move updating mocks toggle to setUp"

This reverts commit 976e90859fce6b10ed38f35347f69ca8e633b7c3.

PascalinDe committed:
docs: remove deprecated comments

PascalinDe committed:
test: add api/config/proxy mock

PascalinDe committed:
test: add missing await on test result

PascalinDe committed:
feat: add linting to tests

PascalinDe committed:
fix: fix linter

PascalinDe committed:
test: add evoting API mock

PascalinDe committed:
test: add missing skip instructions to footer tests

PascalinDe committed:
refactor: use 'strict' mode when compiling tests

PascalinDe committed:
test: add /form/index route tests for navbar/footer

PascalinDe committed:
Merge pull request #76 from c4dt/playwright

Smaller changes, also undo of breaking change in PR #69

PascalinDe committed:
fix: fix misleading page numbering and add pagination tests

PascalinDe committed:
refactor: simplify mocking by removing reliance on HAR files

PascalinDe committed:
fix: fix lint

PascalinDe committed:
style: prettify JSON

PascalinDe committed:
test: add form list display tests

PascalinDe committed:
style: add missing type annotations

PascalinDe committed:
Merge pull request #77 from c4dt/70

Add some tests for formIndex route and remove HAR files

PascalinDe committed:
test: do not actually load page to check URL

PascalinDe committed:
test: assert unauthenticated form list views are correct

PascalinDe committed:
test: fix malformatted test data

PascalinDe committed:
test: admin cannot vote on same form as non-admin

PascalinDe committed:
test: add form list display tests

PascalinDe committed:
style: fix linting

PascalinDe committed:
fix: make sure the pages are properly loaded

PascalinDe committed:
test: add new test data

PascalinDe committed:
test: add initial tests for form editing

PascalinDe committed:
test: move api mocks to separate folders

PascalinDe committed:
test: move evoting mocks

PascalinDe committed:
test: apply code review suggestions

PascalinDe committed:
test: disable lint to make code more readable

PascalinDe committed:
Merge pull request #79 from c4dt/playwright

Automated form list tests

PascalinDe committed:
test: added 2nd admin user for ownership tests

ineiti committed:
Fix the voting-benchmark test

This PR follows updates to Dela to make the benchmark test run again.

ineiti committed:
Rearranging things

Various small changes and refactoring made while speeding up voting.

ineiti committed:
Updating cli tools tests to use new dela

PascalinDe committed:
test: wait until there's no more network traffic on reload

ineiti committed:
Merge pull request #84 from c4dt/update_latest_dela

Updating cli tools tests to use new dela

PascalinDe committed:
chore: fix missing command in set up script

PascalinDe committed:
Merge pull request #88 from c4dt/docker

chore: fix missing command in set up script

PascalinDe committed:
fix: do not show 'Add voters' button to non-owner

PascalinDe committed:
refactor: renamed JSON file to be coherent w/ test file

PascalinDe committed:
refactor: applied code review

PascalinDe committed:
chore: do not run tests in parallel

PascalinDe committed:
fix: fix array index

PascalinDe committed:
fix: 'Cancel' button must only be visible to owner

ineiti committed:
Autogenerating .env

When running ./script/run_docker.sh, automatically generate a .env file

PascalinDe committed:
Merge pull request #89 from c4dt/72

clean up test directory, and initial add form management tests

ineiti committed:
Merge pull request #92 from c4dt/autogenerate_env

Autogenerating .env

ineiti committed:
Merge pull request #82 from c4dt/voting_benchmark

Fix the voting-benchmark test

ineiti committed:
Merge pull request #83 from c4dt/refactoring

Refactoring things

PascalinDe committed:
fix: 'Close' button should only be visible to owner

PascalinDe committed:
fix: 'Combine' button should only be visible to owner

PascalinDe committed:
docs: cleaned up README, removed obsolete files and updated .env.example

PascalinDe committed:
Merge pull request #93 from c4dt/cleanup

docs: cleaned up README, removed obsolete files and updated .env.example

PascalinDe committed:
test: only show 'Delete' button to owner

PascalinDe committed:
test: show 'Shuffle' button only to owner

PascalinDe committed:
fix: show 'Setup' button only to owner

PascalinDe committed:
test: show 'Open' button only to owner

PascalinDe committed:
fix: show 'Initialize' button only to owner

PascalinDe committed:
fix: show 'Decrypt' button only to owner

PascalinDe committed:
fix: reload page to update local caches when authorizations have changed

PascalinDe committed:
Merge pull request #96 from c4dt/playwright

Action Buttons that manipulate a form are only visible to the owner

PascalinDe committed:
test: test that buttons are not visible in form states when they are not active

PascalinDe committed:
test: clean up test messages

PascalinDe committed:
test: add missing 'Canceled' state

PascalinDe committed:
test: add 'Vote' button test cases

PascalinDe committed:
test: add initialization tests

PascalinDe committed:
test: add 'Setup' route calling test

PascalinDe committed:
test: add 'Open' route calling test

PascalinDe committed:
test: add 'Cancel' route calling test

PascalinDe committed:
test: refactor button tests and add tests for remaining buttons

PascalinDe committed:
test: test adding voters

PascalinDe committed:
test: fix voting test

PascalinDe committed:
fix: add missing route mocks

PascalinDe committed:
style: fix linting warnings

PascalinDe committed:
fix: add missing file

PascalinDe committed:
refactor: apply review suggestions

PascalinDe committed:
style: fix linter

PascalinDe committed:
refactor: move auth tools to separate location at root

PascalinDe committed:
Merge pull request #100 from c4dt/72

Add tests for the correct routes of the form management buttons

PascalinDe committed:
fix: non-voter user should not be able to see voting screen

ineiti committed:
Don't let non-form-owner add voters

Also adds a test in the script directory to make sure this doesn't pass anymore.

Closes #81

ineiti committed:
Fix all go version in the tests on 1.19

There is a regression error in go 1.22.
As the go.mod fixes the go version at 1.19, all github workflow now
use go 1.19.

ineiti committed:
Fixing some warnings

I don't know why they show up in WebStorm but not with eslint...

ineiti committed:
Merge pull request #116 from c4dt/fix_go_version

Fix all go version in the tests on 1.20

ineiti committed:
Running crash tests up to three times

The crash tests often fail, but sometimes also run successfully.
This PR runs them 3 times and returns OK if they are run OK at least once.

ineiti committed:
Merge pull request #115 from c4dt/block_admin_changes

Don't let non-form-owner add voters

PascalinDe committed:
test: add tests for handling minimum/maximum/correct number of votes and ballot display

PascalinDe committed:
Merge pull request #117 from c4dt/99

Non-voter user should not see the ballot

PascalinDe committed:
fix: fix test w/ non-unique identifier

PascalinDe committed:
Merge pull request #120 from c4dt/playwright

Add remaining ballots tests

ineiti committed:
Merge pull request #119 from c4dt/rerun_crash_tests

Running crash tests up to three times

ineiti committed:
Merge pull request #118 from c4dt/fix_warnings

Fixing some warnings

PascalinDe committed:
style: remove unused JSON file

PascalinDe committed:
refactor: add another result

PascalinDe committed:
fix: fix result labels

PascalinDe committed:
test: add tests individual/grouped results

PascalinDe committed:
test: add more detailed forms

PascalinDe committed:
feat: show total count instead of percentage and order by count

PascalinDe committed:
Merge pull request #123 from c4dt/45

Display total count instead of percent and order by count

PascalinDe committed:
feat: update smart contract

PascalinDe committed:
feat: add URL field to choices

ineiti committed:
Putting ballots in separate memory

This PR separates the ballots from the forms.
This allows to run adding a new ballot much faster.
When there are more than 100 ballots, this gets very important:
adding a 1000th ballot can take 1s, a 10'000th ballot 10s.

Using this PR, up to the 10'000th ballot it only takes 100ms.

ineiti committed:
Improve speed of docker build

ineiti committed:
Add voting script to the backend

This also updates some of the outdated settings in the frontend

ineiti committed:
Putting FormFromStore into a single place

ineiti committed:
Correctly store and retrieve DKGs

ineiti committed:
Comments and error handling

ineiti committed:
Synching is now done in dela

ineiti committed:
Making tests work

ineiti committed:
Making tests pass

PascalinDe committed:
fix: fix internalisation problems w/ text results

PascalinDe committed:
refactor: refactor choice display

PascalinDe committed:
style: ran prettier

ineiti committed:
Merge pull request #121 from c4dt/separate_ballots

Putting ballots in separate memory

ineiti committed:
Point to latest c4dt/dela commit

ineiti committed:
Merge pull request #132 from c4dt/e2e_fixes

E2e fixes

PascalinDe committed:
Merge pull request #133 from c4dt/37

Add URL for each candidate

PascalinDe committed:
feat: add EPFL logo w/ link to EPFL homepage

PascalinDe committed:
feat: use Arial font

PascalinDe committed:
feat: change bg color to EPFL style

PascalinDe committed:
feat: change fg color to EPFL style

PascalinDe committed:
feat: add URL to smart contract

PascalinDe committed:
feat: add title URL to frontend

ineiti committed:
Following latest dela main and fixing local_forms.sh with new JSON format

ineiti committed:
Merge pull request #136 from c4dt/update_dela

Following latest dela main and fixing local_forms.sh with new JSON format

PascalinDe committed:
feat: use EPFL link style

PascalinDe committed:
fix: fix missing Title field when not 'en' language

PascalinDe committed:
fix: fix missing title link

PascalinDe committed:
test: add EPFL logo test

PascalinDe committed:
fix: fix missing build stage

PascalinDe committed:
Merge pull request #137 from c4dt/docker

Fix missing build stage

PascalinDe committed:
Merge pull request #135 from c4dt/42

EPFL corporate design

PascalinDe committed:
fix: fix inverted hyperlink colours

PascalinDe committed:
fix: fix English not displaying by default for Title fields

ineiti committed:
Merge pull request #139 from c4dt/frontend

Minor fixes

PascalinDe committed:
fix: fix unhandled update

PascalinDe committed:
fix: fix color scheme

PascalinDe committed:
Merge pull request #140 from c4dt/frontend

Minor frontend fixes

ineiti committed:
Adding benchmark tests

Closes #47

PascalinDe committed:
feat: add AdditionalInfo field to smart contract

PascalinDe committed:
feat: add AdditionalInfo field to frontend

ineiti committed:
Merge pull request #146 from c4dt/add_speeds

Adding benchmark tests

PascalinDe committed:
Merge pull request #147 from c4dt/141

Add AdditionalInfo field

ineiti committed:
adding versions

This shows the current version of the build

PascalinDe committed:
fix: add missing JSON/schema validation

PascalinDe committed:
Merge pull request #149 from c4dt/frontend

fix: add missing JSON/schema validation

ineiti committed:
Merge pull request #148 from c4dt/version_display

adding versions

ineiti committed:
Put throw outside of try/catch

ineiti committed:
Merge pull request #150 from c4dt/increase_timeout

Increase timeout

multiscan committed:
[fix] when directory is a symlink

multiscan committed:
[feat] chunked voter sciper upload

multiscan committed:
[doc] explain why we add sendVoters async function

PascalinDe committed:
style: fixed linting

PascalinDe committed:
feat: add error messages

PascalinDe committed:
feat: add loading animation when adding voters

PascalinDe committed:
feat: use translations for error messages

PascalinDe committed:
fix: clean-up error handling

PascalinDe committed:
fix: remove outdated comment

PascalinDe committed:
fix: add missing  call

PascalinDe committed:
style: apply linting

PascalinDe committed:
fix: fix SCIPER verification and add it to adding policy functions

PascalinDe committed:
feat: do not link to form management view for non-admin users

PascalinDe committed:
feat: require authentication as admin to access form management view

PascalinDe committed:
Merge pull request #158 from c4dt/156

Non-admins cannot access form management view

PascalinDe committed:
fix: use Number.isNaN

PascalinDe committed:
fix: add missing await statements

PascalinDe committed:
Merge pull request #157 from c4dt/feat_batch_voters

Add voters in batches

dragonleman committed:
Fix French language: at most -> au plus

PascalinDe committed:
Merge pull request #160 from c4dt/fix-french-language-at-most-au-plus

Fix French language: at most -> au plus

ineiti committed:
Updating scripts

Add the latest changes to the frontend API to the script.
Also removes the setup of the certificates, which is not needed for grpc.

ineiti committed:
Merge pull request #161 from c4dt/update_scripts

Updating scripts

PascalinDe committed:
feat: clarify size computation

PascalinDe committed:
fix: use length of encoded ID to determine total length of ballot

PascalinDe committed:
feat: throw error if encoded ballot is longer than maximum ballot size expected by the backend

PascalinDe committed:
docs: updated documentation on ballot encoding

PascalinDe committed:
Merge pull request #168 from c4dt/167

fix: use size of Base64-encoded ID to calculate ballot length for decryption

PascalinDe committed:
feat: change default filter on form list to 'Open'

PascalinDe committed:
Merge pull request #169 from c4dt/change-default-filter

feat: change default filter on form list to 'Open'

ineiti committed:
Only pushing if merge on main

With this PR, the docker images are built, but only pushed when the
action is run on the 'main' branch.

ineiti committed:
Merge pull request #171 from c4dt/test_docker_build

Only pushing if merge on main

PascalinDe committed:
feat: hide elections on blocklist

PascalinDe committed:
refactor: minor code clean-up

Co-authored-by: Linus Gasser <linus.gasser@epfl.ch>

PascalinDe committed:
Merge pull request #173 from c4dt/hide-elections

feat: hide elections on blocklist

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dedis/d-voting?pullRequest=389) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/dedis/d-voting?pullRequest=389) before we can accept your contribution.<br/>**2** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: ineiti<br/>:white_check_mark: PascalinDe<br/>:x: dragonleman<br/>:x: multiscan<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dedis/d-voting?pullRequest=389) it.</sub>

PascalinDe committed:
fix: add '-y' flag to make sure apt builds autonomously

ineiti committed:
Merge pull request #180 from c4dt/fix-docker

fix: add '-y' flag to make sure apt builds autonomously

ineiti committed:
@Pierluca's and @jbsv's comments

- change name from SuffragiaBlock to SuffragiaBatch
- add prefixed.Snapshot
- added issue to t.Skip in TestHandler_StartShuffle
- added more issues

ineiti committed:
Moving paths from github.com/c4dt to github.com/dedis

ineiti committed:
Merge pull request #181 from c4dt/comments_dedis

@Pierluca's and @jbsv's comments
